src/Morphir/JsonSchema/Backend.elm

```diff
@@ -105,55 +111,178 @@ extractTypes modName definition =
+
+                ( "Morphir.SDK:Result:result", [ error, value ] ) ->
+                    [ mapType error
+                        |> Result.map
+                            (\errorSchema ->
+                                Array (TupleType [ Const "Err", errorSchema ] 2) True
+                            )
+                    , mapType value
+                        |> Result.map
+                            (\valueSchema ->
+                                Array (TupleType [ Const "Ok", valueSchema ] 2) True
+                            )
+                    ]
+                        |> ResultList.keepFirstError
+                        |> Result.map OneOf
+
+                ( "Morphir.SDK:Dict:dict", [ keyType, valueType ] ) ->
+                    let
+                        tupleSchemaList =
+                            [ mapType keyType, mapType valueType ]
+                    in
+                    tupleSchemaList
+                        |> ResultList.keepFirstError
+                        |> Result.map (\tupleSchema -> Array (ListType (Array (TupleType tupleSchema 2) False)) True)
 
                 _ ->
-                    Nothing
-
-        Type.Record a fields ->
+                    Ok
+                        (Ref
+                            (String.concat
+                                [ "#/$defs/"
+                                , moduleName |> Path.toString Name.toTitleCase "."
+                                , "."
+                                , localName |> Name.toTitleCase
+                                ]
+                            )
+                        )
+
+        Type.Record _ fields ->
             fields
-                |> List.foldl
-                    (\field maybeDictSoFar ->
-                        maybeDictSoFar
-                            |> Maybe.map2
-                                (\schemaType dictSofar ->
-                                    Dict.insert (Name.toTitleCase field.name) schemaType dictSofar
-                                )
-                                (mapType field.tpe)
+                |> List.map
+                    (\field ->
+                        mapType field.tpe
+                            |> Result.map (\fieldSchemaType -> ( Name.toCamelCase field.name, fieldSchemaType ))
+                    )
+                |> ResultList.keepFirstError
+                |> Result.map (Dict.fromList >> Object)
+
+        Type.Tuple _ typeList ->
+            typeList
+                |> List.map
+                    (\tpe ->
+                        mapType tpe
+                    )
+                |> ResultList.keepFirstError
+                |> Result.map
+                    (\itemType ->
+                        Array (TupleType itemType (typeList |> List.length)) False
                     )
-                    (Just Dict.empty)
-                |> Maybe.map Object
 
         _ ->
-            Nothing
+            Err "Cannot map this type"
```

Comment:
Can you please add the actual type that we couldn't map into the message? That will make it much easier to find out what went wrong.